File: bubble_sort.js
function bubbleSort(array){
	var did_sort = 1;
	var temp;
	while(did_sort > 0){
		did_sort = 0;
		for(var i = 0; i < array.length-1; i++){
			if(array[i] > array[i+1]){
				temp = array[i];
				array[i] = array[i+1];
				array[i+1] = temp;
				did_sort++;
			}
		}
	}
	return array;
}


var myArray = [5,4,3,2,1];
bubbleSort(myArray);
console.log(myArray);

File: bubble_sort.rb
def bubble_sort(arr)

	did_sort = true
	while did_sort
		did_sort = false

		arr.length.times do |index|
			if index < arr.length-1 && arr[index] > arr[index+1]
				 arr[index], arr[index+1] = arr[index+1], arr[index]
				 did_sort = true
			end
		end


	end
	arr
end




arr = Array.new(10) { rand(1..99) }


puts "before #{arr}"
bubble_sort(arr)
puts "after #{arr}"

